modules/sources/contracts/v3/fetch-unwrap-rates.ts

```diff
@@ -0,0 +1,24 @@
+import { Chain } from '@prisma/client';
+import { Multicaller3Viem } from '../../../web3/multicaller-viem';
+import MinimalErc4626Abi from '../abis/MinimalERC4626';
+import { formatEther, parseEther } from 'viem';
+
+/**
+ * Fetches convertToAssets rates for a list of ERC4626 tokens and returns them as floats
+ *
+ * @param addresses
+ * @param chain
+ * @returns
+ */
+export const fetchUnwrapRates = async (addresses: string[], chain: Chain) => {
+    const caller = new Multicaller3Viem(chain, MinimalErc4626Abi);
+    addresses.forEach((address) => caller.call(address, address, 'convertToAssets', [parseEther('1')]));
+    const results = await caller.execute<{ [id: string]: bigint }>();
+
+    // Convert the results to floats
+    const formattedResults = Object.fromEntries(
+        Object.entries(results).map(([key, value]) => [key, parseFloat(formatEther(value))]),
```

Comment:
I'd rather not cast them to floats to prevent losing precision on the rate. Is there an issue on using strings?
I can update them on my end, just let me know if you think we might have an issue with that change.

Reply:
hey, sure, whatever works. the only potential problem is that DB always stores floats, so breaking the standard will be confusing. would it make sense to change other values to keep evm scaling as well? eg: price rates from rate providers?

Reply:
I'd say we should change all rates to strings 👍 
We're doing our best to get balancer-maths and SOR to 100% match against on-chain results, so if we're able to avoid losing precision, I'd rather go that way.
Is there any downside to storing strings other than having to migrate everything?

Reply:
i'd expect the same goes for token balances? totally agree on data correctness, though the decision, made ages  ago, runs deep, impacting things like the subgraph. SOR is a pretty good reason to have evm values. I think we can start adding native values without much or tiny impact on existing code. tagging @franzns for thoughts.

Reply:
Yeah, we could have a separate issue to keep track of this refactor, but I'd consider it low priority at the moment. Anyway, my suggestion is to already make new ones string/evm scale and even refactor as we touch them on other PRs, rather than keep creating floats and refactor everything at once later.
But that's my opinion.
Totally fine if you guys disagree and prefer to keep the standard (floats) for now and refactor all at once in a single PR.

Reply:
Just so we have it registered here as well -> I don't think we need to store amounts in `evm scale`, I just think we need to avoid `floats` and store `human scale` as `strings`.

Reply:
doesn't convertion from evm to human scale loose precision already?

Reply:
not necessarily - as long as we store as strings it should be ok - for example:
- evm scale (18 decimals): 12345678901234567890
- human scale (string): 12.345678901234567890
- human scale (float): 12.34567890123457

Ps: I don't remember exactly how many decimals float support, but it will eventually truncate

Reply:
btw - `parseFloat` does lose precision, but that's why we shouldn't use it 🙃
you have to use `parseUnits` or `parseEther`

Reply:
yes, exactly. good summary, thanks!